frontend/src/components/PostDetail.js

```diff
@@ -4,22 +4,26 @@ import { Link } from 'react-router-dom'
 
 export default function PostDetail ({ postid, posts, comments }) {
     return (
-        <div>  
+        <div>
             <h2>Here is the post!</h2>
-            <h3>{posts[postid].title}</h3>
-            <p>{posts[postid].body}</p>
-            <p>Category: {posts[postid].category}</p>
-            <p>Author: {posts[postid].author}</p>
-            <p>Votes: {posts[postid].voteScore}</p>
+      {/* The id of the posts from the database do not match with the react router id */}
+            <h3>{/*posts[postid].title*/}</h3>
+            <p>{/*posts[postid].body*/}</p>
+            <p>Category: {/*posts[postid].category*/}</p>
+            <p>Author: {/*posts[postid].author*/}</p>
+            <p>Votes: {/*posts[postid].voteScore*/}</p>
             <h2>Here are the comments!</h2>
-            {comments.filter((comment)=> comment.parentId === posts[postid]['id']).map((comment, i) => (
+      {/* fails when comments are empty */}
+            {/*comments.filter((comment)=> comment.parentId === posts[postid]['id']).map((comment, i) => (
                 <div key={i}>
                     <Comment comment={comment} />
                 </div>
```

Comment:
commented these out because you have an issue with the post id. when you create a post you use a "cuid" but in the router they start from 0 which is being used in this view which is throwing an error.